webapp/src/pages/Settings.tsx

```diff
@@ -31,29 +62,540 @@ const Settings: React.FC = () => {
 
   const resultingConfig = { ...data, ...partialConfig };
 
-  const switchNullOrDefault = (field: keyof AzimuthConfig) => (
+  const divider = <Divider sx={{ margin: 1 }} />;
+
+  const displaySectionTitle = (section: string) => (
+    <Box sx={{ m: 1, width: "20ch" }}>
+      <Typography textTransform="capitalize" variant="caption" fontSize={16}>
+        {section}
+      </Typography>
+    </Box>
+  );
+
+  const displaySubSectionTitle = (section: string) => (
+    <Box sx={{ m: 1, width: "20ch" }}>
+      <Typography textTransform="capitalize" variant="caption" fontSize={14}>
+        {section}
+      </Typography>
+    </Box>
+  );
+
+  const displayToggleSectionTitle = (
+    config: keyof AzimuthConfig,
+    section: string = config
+  ) => (
+    <Box sx={{ m: 1, width: "20ch" }}>
+      {switchNullOrDefault(config, Boolean(resultingConfig[config]))}
+      <Typography textTransform="capitalize" variant="caption" fontSize={15}>
+        {section}
+      </Typography>
+    </Box>
+  );
+
+  const displayPostprocessorToggleSection = (
+    pipeline: PipelineDefinition,
+    pipelineIndex: number
+  ) => (
+    <Box sx={{ m: 1, width: "20ch" }}>
+      <Checkbox
+        sx={{ paddingLeft: 0 }}
+        size="small"
+        checked={Boolean(pipeline.postprocessors)}
+        disabled={isError || isFetching}
+        onChange={(event) => {
+          setPartialConfig({
+            ...partialConfig,
+            pipelines: _.unionBy(
+              [
+                {
+                  ...pipeline,
+                  postprocessors: event.target.checked ? [] : null,
```

Comment:
The `Checkbox`'s `onChange` handler provides a second argument which is directly the `checked` boolean, so no need to mess with `event.target`:
```suggestion
        onChange={(_, checked) => {
          setPartialConfig({
            ...partialConfig,
            pipelines: _.unionBy(
              [
                {
                  ...pipeline,
                  postprocessors: checked ? [] : null,
```

Reply:
Okay, thank you for the suggestion here. I had to write it that way because the "_" confuse between onchange attribute and lodash attributes.

Reply:
Of course! I think that's a cool trick with the `...[]`. This is my most up-voted [answer in stack overflow](https://stackoverflow.com/questions/32197927/standard-conventions-for-indicating-a-function-argument-is-unused-in-javascript/58738236#58738236).

If you're curious, another neat solution is to destructure the unused argument into nothing with `{}`. However, this is less general as it doesn't work if the argument can be `undefined` or `null`. So, I prefer `...[]`.
```diff
-        onChange={(...[, checked]) => {
+        onChange={({}, checked) => {
```